src/dataset.py

```diff
@@ -131,3 +172,147 @@ def get_other_files(self):
 
     def get_interactome(self):
         return self.interactome.copy()
```

Comment:
We may need this to be a deep copy now that we are manipulating the interactome differently for each algorithm.